node/src/components/linear_chain/pending_signatures.rs

```diff
@@ -41,7 +59,19 @@ impl PendingSignatures {
         let pending_sigs = self
             .pending_finality_signatures
             .values_mut()
-            .filter_map(|sigs| sigs.remove(&block_hash))
+            .filter_map(|sigs| {
+                // Collect the signature only if it's been verified already.
+                if let Some(sig_status) = sigs.get(block_hash) {
+                    if sig_status.is_bonded() {
+                        sigs.remove(block_hash)
+                    } else {
+                        None
+                    }
+                } else {
+                    None
+                }
```

Comment:
```suggestion
                match sigs.get(block_hash) {
                    Some(sig_status) if sig_status.is_bonded() => {
                        sigs.remove(block_hash)
                    }
                    _ => None
                }
```

Reply:
Done in [`61a1ac0` (#1291)](https://github.com/CasperLabs/casper-node/pull/1291/commits/61a1ac07350ed98efaa052725463588407f7761d)